packages/lesswrong/components/editor/EditorFormComponent.tsx

```diff
@@ -435,6 +454,24 @@ export const EditorFormComponent = ({
 
   const actualPlaceholder = ((collectionName === "Posts" && getPostPlaceholder(document)) || editorHintText || hintText || placeholder);
 
+  useEffect(() => {
+    if (!isCollabEditor && collectionName === 'Posts' && fieldName === 'contents') {
+      setAutosaveEditorState((_) => () => new Promise((resolve, reject) => {
```

Comment:
I think this could be
```
setAutosaveEditorState((_) => async () => {
```
which would be a lot less confusing?

Reply:
Do you mean this?

```
      setAutosaveEditorState((_) => async () => {
        if (editorRef.current && shouldSubmitContents(editorRef.current)) {
          const { originalContents: { type, data: value } } = await editorRef.current.submitData();
          return await saveRemoteBackup({ type, value });
        }
      });
```

I don't remember why I wrote it that way originally... maybe something related to the setState?